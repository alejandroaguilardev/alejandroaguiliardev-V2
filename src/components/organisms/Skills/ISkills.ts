export interface ISkills {
	title: string;
	description: string;
	dataSkills: {
		title: string;
		subtitle: string;
		description: string;
		icon: string;
	}[];
}
