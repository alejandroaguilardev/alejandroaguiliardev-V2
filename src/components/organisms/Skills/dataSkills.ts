export const dataSkills = [
	{
		title: "JAVASCRIPT",
		subtitle: "",
		description: "I perform very well in this programming language.",
		icon: "/img/skills/javascript.svg",
	},
	{
		title: "TYPESCRIPT",
		subtitle: "",
		description: "Moderate experience in this programming language.",
		icon: "/img/skills/typescript.svg",
	},
	{
		title: "PHP",
		subtitle: "",
		description: "I perform very well in this programming language.",
		icon: "/img/skills/php.svg",
	},
	{
		title: "SOLIDITY",
		subtitle: "",
		description: "Programming language that I master to create smart contracts",
		icon: "/img/skills/solidity.svg",
	},
	{
		title: "NEXTJS",
		subtitle: "",
		description: "I perform very well in this Framework Front End",
		icon: "/img/skills/nextjs.svg",
	},
	{
		title: "LARAVEL",
		subtitle: "",
		description: "I perform very well in this Framework Back End",
		icon: "/img/skills/laravel.svg",
	},
	{
		title: "NODEJS",
		subtitle: "",
		description:
			"Experience working with Node js Express performing FULL REST API",
		icon: "/img/skills/nodejs.svg",
	},
	{
		title: "MONGODB",
		subtitle: "",
		description:
			"Better performance on this NOSQL database.",
		icon: "/img/skills/mongodb.svg",
	},
];
