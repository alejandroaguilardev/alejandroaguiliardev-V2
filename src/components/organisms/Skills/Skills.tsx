import { Grid, Container } from "@mui/material"
import { TitleSection } from "../../atoms"
import { CardItem } from "../CardItem/CardItem"
import { ISkills } from "./ISkills"

export const Skills = ({ title, description, dataSkills }: ISkills) => {
    return (
        <Container maxWidth="lg" >
            <TitleSection
                title={title}
                description={[description]}
            />

            <Grid container spacing={5} >
                {dataSkills.map((data) => (
                    <Grid item xs={6} md={3} key={data.title}>
                        <CardItem
                            title={data.title}
                            subtitle={data.subtitle}
                            description={data.description}
                            icon={data.icon}
                        />
                    </Grid>
                ))}
            </Grid>
        </Container >
    )
}
