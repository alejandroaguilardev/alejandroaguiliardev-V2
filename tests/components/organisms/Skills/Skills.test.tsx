import { render, screen } from "@testing-library/react";
import { Skills } from "../../../../src/components/organisms/Skills/Skills";

describe("Skills Test", () => {
	const title = "title";
	const description = "description";
	const dataSkills = [
		{
			title: "data title",
			subtitle: "data subtitle",
			description: "data description",
			icon: "/img/banner.jpg",
		},
	];
	test("toMatchSnapshot", () => {
		const { container } = render(
			<Skills title={title} description={description} dataSkills={dataSkills} />
		);
		expect(container).toMatchSnapshot();
		expect(screen.findByText(title)).toBeTruthy();
		expect(screen.findByText(description)).toBeTruthy();
		expect(screen.findByText(dataSkills[0].title)).toBeTruthy();
	});
});
