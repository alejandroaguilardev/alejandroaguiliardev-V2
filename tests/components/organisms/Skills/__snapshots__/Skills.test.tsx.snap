// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skills Test toMatchSnapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-1yjvs5a"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-162zxzj-MuiTypography-root"
        role="title"
      >
        title
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1mqboxc-MuiTypography-root"
        role="description"
      >
        description
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-3 css-1unik2p-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-15cnh1e-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-18i215d"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2770%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="data title"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="MuiBox-root css-gajhq5"
          >
            <h3
              class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-yg6atj-MuiTypography-root"
            >
              data title
            </h3>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1a3kej9-MuiTypography-root"
            >
              data subtitle
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-r5yzcc-MuiTypography-root"
            >
              data description
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
